lib/templateFactory.js

```diff
@@ -52,6 +53,36 @@ class TemplateFactory extends BaseFactory {
     }
 
     /**
+     * Get a the latest template by config
+     * @method getTemplate
+     * @param  {Object}     config               Config object
+     * @param  {String}     config.name          The template name
+     * @param  {String}     config.version       Version of the template
+     * @param  {String}     [config.label]       Label of template
+     * @return {Promise}
+     */
+    getTemplate(config) {
+        // get all templates with the same name
+        return super.list({ name: config.name })
+            .then((templates) => {
+                // get templates that have version prefix as config.version
+                const filtered = templates.filter(template =>
+                    template.version.startsWith(config.version));
+
+                // Sort templates by descending order
+                filtered.sort((a, b) => compareVersions(b.version, a.version));
+
+                // If no config.label, return latest one
+                if (filtered.length > 0 && !config.label) {
```

Comment:
For some odd reason, I misinterpreted config.label to be an array. As a String, my concern isn't valid.

This is cool 👍